File: CSA_HW2/Bird.cpp
#include "Bird.h"

//------------------------------------------------------------------------------
// ------------------- CPP FILE OF CHILD CLASS 'BIRD' --------------------------
//------------------------------------------------------------------------------

// Implementation of the method for printing information about bird.
void Bird::Output(std::ofstream &output_file_stream) {
    output_file_stream << "Animal " << name_ << " with weight " << weight_ << " grams. \n";
    output_file_stream << "It's quotient from name to weight is " << Quotient() << ". \n";
    if (is_migratory_ == 0) {
        output_file_stream << "It's a bird and it's not migratory.\n";
    } else {
        output_file_stream << "It's a bird and it's migratory.\n";
    }
}

File: CSA_HW3/Animals/Bird.py
from Animals.Animal import Animal


# ------------------------------------------------------------------------------
# --------------------------- CHILD CLASS 'BIRD' -------------------------------
# ------------------------------------------------------------------------------

class Bird(Animal):

    # Constructor for child class bird with extra parameter - is bird migratory or not.
    def __init__(self, name, weight, is_migratory):
        super(Bird, self).__init__(name, weight)
        self.__is_migratory__ = is_migratory

    # Method for printing information about bird.
    def out(self):
        # Calling method for print basic info about bird.
        info = super(Bird, self).out()
        info += "It's a bird and it's "
        if self.__is_migratory__ == 0:
            info += "not "
        info += "migratory.\n"
        return info
